### PR TITLE
Add first CSV files for the data model

### PR DESCRIPTION
**Pull request by dicortazar**
This commit provides the addition of a small script to
translate from ElasticSearch JSON mapping document to
CSV file

### CODE REVIEW

**Comment by alpgarcia** [comment]
I've added a 'd' to centralize[d]. Sorry for commiting the change directly on the branch ;)

It may be a good idea to make explicit that CSV files could include more columns with other information on how to build fields from RAW indexes, comments, field descriptions or whatever. In fact, although it is not going to be used for testing, defining a set of possible columns would be needed if we want to use these files as a reference for enrichment. Let's agree if this makes sense before approving this PR :)

**Comment by alpgarcia** [comment]
On the other hand, I think it would be a good idea to unify source code style, so maybe @sduenas could also review *.py files and suggest some changes if needed?

**Comment by jgbarah** [comment]
About `schema/README.md`: could we be more specific? Now I read:

> Each of the CSV files contains two main columns: name and type. 

I think we should be more conclusive, like:

> Each of the CSV files contain two columnos
> * name
> * type

In fact, I think we need at least some more:

* For some rows, we would need extra information. For example, for strings, I guess we need to specify if they are analyzed or not. For dates, I think we should specify the date format (or maybe specify that for all date fields we use such and such format).
* We need a description field (brief description of the ElasticSearch field).

**Comment by jgbarah** [comment]
In addition, I would rather split this pull request, puttiing the software into a different pull request. It would be better to agree on the format of the CSV files, and then discuss on software to produce them, than trying to do both together (even if there are no changes to the program because of the discussion about the format).

**Comment by jgbarah** [comment]
@alpgarcia said:
> On the other hand, I think it would be a good idea to unify source code style, so maybe @sduenas could also review *.py files and suggest some changes if needed?

 I agree. As commented above, separating the script into a different pull request would help.

**Comment by jgbarah** [comment]
The description of the commit does not fit what is in the commit. In fact, the title and the text below it show different parts of the same commit. Please, produce a description for the commit which matches what is in it:

> Add first CSV files for the data model
>
>This commit provides the addition of a small script to translate from ElasticSearch JSON mapping document to CSV file

**Comment by jgbarah** [comment]
The commit by @alpgarcia should be squezeed with the first one by @dicortazar.

File: src/json2csv/tests/run_tests.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2015-2016 Bitergia
```
Comment (jgbarah) [inline src/json2csv/tests/run_tests.py]: Very likely copyright is 2017.

File: src/json2csv/tests/test_json2csv.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2015-2016 Bitergia
```
Comment (jgbarah) [inline src/json2csv/tests/test_json2csv.py]: Same about copyright

**Comment by dicortazar** [comment]
Thanks for the comments. Replying to those:

> It may be a good idea to make explicit that CSV files could include more columns by @alpgarcia 

Done. I just mentioned that this could be done in the future.

> On the other hand, I think it would be a good idea to unify source code style, so maybe @sduenas could also review *.py files and suggest some changes if needed? by @alpgarcia 

Let's wait for his comments on the source code

> About schema/README.md: could we be more specific? Now I read... by @jgbarah 

I've simply added that we could add more columns in the near future. We're not using those now and we're not even having those automatically as this set of CSV files. I'd leave this as further work. We're just testing if this could work and fits everyone needs.

> In addition, I would rather split this pull request, puttiing the software into a different pull request by @jgbarah 

The set of CSV files that we already have were produced by the software. I mean, that's a direct translation from the mapping in JSON format to CSV format. No other fields were found in the JSON document, so no other fields were added to the CSV file. The software was merged at the same time than the documentation as the software is basically producing the CSV files and the documentation is written on top of the CSV files format. Again, this is a first step to have CSV files and directly exporting the JSON docs from the current existing indexes sounds like the fastest approach here.
I do not think it's a good idea to have a loooong discussion about formats before producing some software. Indeed we can ignore the software if you prefer.

> The description of the commit does not fit what is in the commit. In fact, the title and the text below it show different parts of the same commit. Please, produce a description for the commit which matches what is in it: by @jgbarah 

A suggestion is welcome. I think it works.

> The commit by @alpgarcia should be squezeed with the first one by @dicortazar. by @jgbarah 

I do not see potential issues about this other than having more commits in the repository. As we're discussing about documentation this tends to be more subjective so more changes are potentially expected. In addition the tool we're using (GitHub in this case) allows us to work in this way.

**Comment by dicortazar** [comment]
> You should split commit in two as @jgbarah says on his previous review. The description of the commit doesn't match with what you are adding.

The current CSV format does not have any sense without the code as the code is the one producing such format.

> One question, why do we need two files JSON and CSV? That means we will have to keep them in sync. That's prone to errors. If CVS are generated by your script. Why do we need to keep them in a repository?

We need those in panels for testing. And this is basically what we discussed. We want to have the data model referenced in any place. Still probably discussing between JSON and CSV (main reason to add both).

**Comment by dicortazar** [comment]
Hey, as there is not a clear process define and any of us has a different view about the format, the software and the color of the bikeshed (^^), please go to #98 and let's have this discussion there. I'll dismiss this branch and produce a new one when there's an agreement.

Whatever we have there as agreement in the issue, that will be implemented so we can properly focus on the CSV files and the software to produce these.

**Comment by alpgarcia** [comment]
My proposal:
* Upload just some basic csvs to begin with. Of course we need to discuss on this afterwards. Right now there will be as mappings, allowing humans to add columns on the right to 'communicate'. Don't mind about them before CSVs are included in our testing infrastructure that would be released next week.

* Afterwards: once we have some very basic tests including CSVs, Panel JSONs and ES Mappings, let's improve CSVs to add those things not covered by a regular ES mapping but needed to build an enriched index. We could add columns just for documenting things and even some more complex columns to improve testing. Anyway, the latter one is out of focus for a first step if we want to keep going.

**Comment by sduenas** [comment]
I agree with @alpgarcia . Just keep it as simple as possible at first.

**Comment by dicortazar** [comment]
@sduenas , @alpgarcia please comment at #98 

Not going forward till this is all clear.

**Comment by jgbarah** [comment]
I agree with @alpgarcia and @sduenas (except minor issues that could be dealt with later on). In short:

* What we need asap are the CSV files, so that we can start testing. If they are produced by some software, that's fine, but not that important. In fact, from now on, it will be the other way around: we start with a csv file, and the produce Python code in GrimoireELK to produce the corresponding index.

* If we want the script, that's fine. But let's open another PR for it, and let's keep this for the format and first CSV files implementing it.

I would add in the description the fields for the "analyzed" thing and (at the very least) for a description, even if they are not still in the CSV, since they provide with a framework for growing (I could write some descriptions for CSV, for example, once we're done with this). But I won't make a cause of it. I will just PR about that once we're done with this.

@dicortazar: could we limit this PR to the description of the format and some first CSVs, and leave the script for other, and if needed the JSON for yet another, and we're done?

**Comment by dicortazar** [comment]
Please let's discuss all of this at #98 CC @alpgarcia @jgbarah @sduenas 
Add any comment there. Whatever we decide there I'll produce a new version and remove this branch.

As it's clear that the four of us have an opinion about the following steps (even the first one), let's discuss at #98 and then implement those conclusions in a new branch.

**Comment by jgbarah** [comment]
As I just said in #98, if you don't want to split this in CSV and software, and remove the JSON files, ok. I don't understand it but OK, we need this asap. We can followup the discussion latter. So please consider my last comment to be as follows:

> What we need asap are the CSV files, so that we can start testing. In fact, from now on we will start with a csv file, and the produce Python code in GrimoireELK to produce the corresponding index. So let's accept the format and the first CSV files implementing it asap. If you insist in including the script as well, let's do it that way. I don't see the need for the JSON files for now, but still, if you insist, let's do it that way too. I just want my CSV files in ;-)

This said, if we could at least add the description field in the format as well, you would make my day ;-)

**Comment by dicortazar** [comment]
Thanks for your comments.

Closing this PR as agreed in #98 and opening a new branch with just the CSV files.

**Pull request closed without merging**